packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -50,6 +63,76 @@ const TextStyle _kLargeTitleTextStyle = TextStyle(
   color: CupertinoColors.black,
 );
 
+// There's a single tag for all instances of navigation bars because they can
+// all transition between each other (per Navigator) via Hero transitions.
+const _HeroTag _heroTag = _HeroTag();
+
+class _HeroTag {
+  const _HeroTag();
+  // This class exists because the Gallery smoke test wants all widgets from
+  // debugDumpApp() to have a toString();
+  @override
+  String toString() => "Cupertino navigation bars' Hero tag";
+}
+
+TextStyle _navBarItemStyle(Color color) {
+  return new TextStyle(
+    fontFamily: '.SF UI Text',
+    fontSize: 17.0,
+    letterSpacing: -0.24,
+    color: color,
+  );
+}
+
+/// Returns `child` wrapped with background and a bottom border if background color
+/// is opaque. Otherwise, also blur with [BackdropFilter].
+Widget _wrapWithBackground({
+  Border border,
+  Color backgroundColor,
+  Widget child,
+  bool annotate = true,
```

Comment:
nit: "annotate" doesn't convey what exactly the annotation does. Maybe `affectSystemUi` or some such?